tests/NeoServer.Game.Creatures.Tests/WalkableCreature/PlayerTest.cs

```diff
@@ -296,4 +299,136 @@ public void PlayerDoesNotGainSkillsWhenUsingAHealingRune()
+
+        var sut = PlayerTestDataBuilder.Build(hp: 100, speed: 300, pathFinder: pathFinder);
+        var stoppedWalkEventEmitted = false;
+
+        sut.OnStoppedWalking += _ => stoppedWalkEventEmitted = true;
+
+        var creature = new Mock<ICreature>();
+        creature.Setup(x => x.Location).Returns(new Location(100, 105, 7));
+        creature.Setup(x => x.CreatureId).Returns(123);
+
+        var tile = map[100, 100, 7] as IDynamicTile;
+
+        sut.SetCurrentTile(tile);
+
+        sut.Follow(creature.Object);
+
+        //act
+        sut.StopAllAction();
+
+        //assert
+        Assert.False(sut.HasNextStep);
+        Assert.False(sut.IsFollowing);
+        Assert.Null(sut.Following);
+        Assert.False(sut.Attacking);
+        Assert.True(stoppedWalkEventEmitted);
+        Assert.Equal(Direction.None, sut.GetNextStep());
+    }
+
+    [Fact]
+    public void Stop_All_Actions_When_IsWalking()
+    {
+        var directions = new[] { Direction.North, Direction.East };
+        var pathFinder = new Mock<IPathFinder>();
+        pathFinder.Setup(x => x.Find(It.IsAny<ICreature>(), It.IsAny<Location>(), It.IsAny<FindPathParams>(),
+            It.IsAny<ITileEnterRule>(), out directions)).Returns(true);
+
+        var sut = PlayerTestDataBuilder.Build(hp: 100, speed: 300, pathFinder: pathFinder.Object);
+
+        var stoppedWalkEventEmitted = false;
+
+        sut.OnStoppedWalking += _ => stoppedWalkEventEmitted = true;
+
+        var tile = new Mock<IDynamicTile>();
+        tile.Setup(x => x.Ground.StepSpeed).Returns(100);
+        tile.Setup(x => x.Location).Returns(new Location(100, 100, 7));
+
+        sut.SetCurrentTile(tile.Object);
+        //First instruction walking
+        sut.WalkTo(new Location(105, 100, 7));
+
+        sut.StopAllAction();
+
+        Assert.False(sut.HasNextStep);
+        Assert.False(sut.IsFollowing);
+        Assert.Null(sut.Following);
+        Assert.False(sut.Attacking);
+        Assert.True(stoppedWalkEventEmitted);
+        Assert.Equal(Direction.None, sut.GetNextStep());
+    }
+
+    [Fact]
+    public void Stop_All_Actions_When_Attacking()
+    {
+        var directions = new[] { Direction.North, Direction.East };
+        var pathFinder = new Mock<IPathFinder>();
+        pathFinder.Setup(x => x.Find(It.IsAny<ICreature>(), It.IsAny<Location>(), It.IsAny<FindPathParams>(),
+            It.IsAny<ITileEnterRule>(), out directions)).Returns(true);
+
+        var sut = PlayerTestDataBuilder.Build(hp: 100, speed: 300, pathFinder: pathFinder.Object);
+
+        var stoppedWalkEventEmitted = false;
+
+        sut.OnStoppedWalking += _ => stoppedWalkEventEmitted = true;
+
+        var tile = new Mock<IDynamicTile>();
+        tile.Setup(x => x.Ground.StepSpeed).Returns(100);
+        tile.Setup(x => x.Location).Returns(new Location(100, 100, 7));
+
+        sut.SetCurrentTile(tile.Object);
+        //First instruction walking
+        sut.WalkTo(new Location(105, 100, 7));
```

Comment:
He we don't need WalkTo and instead Attack

Reply:
This method is not use, i will remove